### PR TITLE
Add Data Table chart support

### PR DESCRIPTION
**Pull request by greatestusername-splunk**
Noticed that table charts were not implemented. 

Added table chart resource
Added test for table chart resource
Added to provider

[Output of test plan, apply, delete plan, delete apply.](https://docs.google.com/document/d/1u7NYOTz7iwkngX_nG2ZActtXIRnpwokzWPT43FaR974/edit?usp=sharing)

@bgola-signalfx

### CODE REVIEW

**Comment by bgola-signalfx** [comment]
You may want to update CHANGELOG too. Other than than LGTM!

**Comment by greatestusername-splunk** [comment]
> 
Changelog updated as requested

File: website/docs/r/table_chart.html.markdown [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    program_text            = "A = data('cpu.usage.total').publish(label='CPU Total')"
+    group_by                = ["ClusterName"]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported in the resource block:
+
+* `name` - (Required) Name of the text note.
```
Comment (tmutdosch) [inline website/docs/r/table_chart.html.markdown]: 'text note' should be Table Chart.

File: website/docs/r/table_chart.html.markdown [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+* `program_text` - (Required) The SignalFlow for your Data Table Chart
+* `description` - (Optional) Description of the text note.
```
Comment (tmutdosch) [inline website/docs/r/table_chart.html.markdown]: 'text note' should be Table Chart.

File: website/docs/r/table_chart.html.markdown (hunk 1 above)
Comment (greatestusername-splunk) [reply website/docs/r/table_chart.html.markdown]: Nice find thanks much!!

**Comment by tmutdosch** [comment]
This looks good to me from the Table Chart API standpoint.

**Review by bgola-signalfx — APPROVED** [review]

**Pull request merged**